src/main/java/frc/robot/commands/autos/components/DriveDistance.java

```diff
@@ -0,0 +1,60 @@
+package frc.robot.commands.autos.components;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.WaitUntilCommand;
+import frc.robot.RobotContainer.MatchState;
+import frc.robot.subsystems.drive.DriveSubsystem;
+
+/** drives in a direction at a given speed for a certain distance */
+public class DriveDistance extends SequentialCommandGroup {
+  double distanceTraveledMeters = 0;
+  
+  Pose2d startPose;
+  Translation2d direction;
+  Translation2d speeds;
+  double distanceToTravel;
+  MatchState matchState;
+
+  /**
+   * @param normSpeed should be in [0,1]
+   */
+  public DriveDistance(
+      DriveSubsystem drive,
+      double normSpeed,
+      double XMeters,
+      double YMeters,
+      boolean fieldRelative) {
+
+    Translation2d desiredTranslation = new Translation2d(XMeters, YMeters);
+
+    this.distanceToTravel = desiredTranslation.getNorm();
+    this.direction = desiredTranslation.div(distanceToTravel);
+    this.speeds = direction.times(normSpeed);
+
+    addCommands(
+        new InstantCommand(
+            () -> {
+              this.startPose = drive.getPose();
+            }),
+        new InstantCommand(
+                () -> {
+                  final double NOT_ROTATING = 0.0;
+                  double xDriveSpeed = matchState.blue ? speeds.getX() : -speeds.getX();
+                  double yDriveSpeed = matchState.blue ? speeds.getY() : -speeds.getY();
```

Comment:
Since we are using 2023 coordinate system (see Basel's comment [here](https://github.com/cachemoney8096/2024-crescendo/pull/25#discussion_r1486977944)), you do not need these ternary operators and should just do positive `getX()` and `getY()`

Reply:
The fact that we're doing this is what causes us to be in the 2023 coordinate system

Reply:
ope yeah good point